### PR TITLE
Reviewed Type Safety in ModManager and World Factory

### PR DESCRIPTION
**Pull request by RustyPotato**
I am still getting familiar with git and other people. So, I decided to handle all the little things. CLion has a handy dandy feature to point out these little casts which might not fit, so I concentrated on those today.

### CODE REVIEW

**Comment by BevapDin** [comment]
CLion probably thinks `std::vector<T>::size()` returns an `unsigned long`, which might be correct for your system, but is not necessarily correct for other systems. On my system, it returns a `unsigned int`. If you want type safety, use `size_t`, `std::size_t`, or `std::vector<T>::size_type` (usually a typedef for `std::size_t`) for values that are used as indices in a vector.

File: src/worldfactory.cpp [hunk 1]
```diff
@@ -144,21 +144,22 @@ WORLDPTR worldfactory::make_new_world( bool show_prompt )
...
+        size_t lasttab; // give placement memory to menus, sorta.
+        const size_t numtabs = tabs.size();
+        while ((unsigned long)curtab < (unsigned long)numtabs) {
             lasttab = curtab;
             draw_worldgen_tabs(wf_win, curtab);
             curtab += tabs[curtab](wf_win, retworld);
 
-            if (curtab < 0) {
+            // If it is -1, or for unsigned size_t, it would be max.
+            if (curtab == (size_t)-1) {
```
Comment (kevingranade) [inline src/worldfactory.cpp]: Please take another approach to addressing this, casting a negative number to an unsigned number and relying on unsigned underflow behavior is far more confusing than having a few warnings floating around.

File: src/worldfactory.cpp [hunk 2]
```diff
@@ -767,12 +772,14 @@ void worldfactory::show_active_world_mods( const std::vector<std::string> &world
 
         if( action == "UP" ) {
             cursor--;
-            if( cursor < 0 ) {
+            // If it went under 0, loop back to the end of the list.
+            if( cursor > num_mods -1 ) {
```
Comment (kevingranade) [inline src/worldfactory.cpp]: Again, triggering unsigned underflow is not a valid approach, if a variable is going to be subtracted below zero, it should be signed.

**Comment by RustyPotato** [comment]
Ok. I have tried for a little while, can anyone explain what gorgon is failing and what I need to do in order to resolve this?

**Comment by Night-Pryanik** [comment]
Jenkins rebuild

**Pull request merged**